### PR TITLE
Laboratory work #1, Polina Kobyakova - 22FPL2

### PR DESCRIPTION
**Pull request by kobyakovapolina**

### CODE REVIEW

File: lab_1_classify_by_unigrams/main.py [hunk 1]
```diff
@@ -11,7 +11,22 @@ def tokenize(text: str) -> list[str] | None:
     :param text: a text
     :return: a list of lower-cased tokens without punctuation
     """
+    en_text = ''
 
+if isinstance(en_text, str):
```
Comment (artyomtugaryov) [inline lab_1_classify_by_unigrams/main.py:16]: Why is it outside the function?

File: main.py [hunk 2]
```diff
@@ -0,0 +1 @@
+print("hello world1")
```
Comment (artyomtugaryov) [inline main.py:1]: remove the file

**Review by artyomtugaryov — CHANGES_REQUESTED** [review]
Reviewed, but hope see more later

File: main.py (hunk 2 above)
Comment (kobyakovapolina) [reply main.py:1]: File was removed

File: lab_1_classify_by_unigrams/main.py (hunk 1 above)
Comment (kobyakovapolina) [reply lab_1_classify_by_unigrams/main.py:16]: I fixed the functiton

File: lab_1_classify_by_unigrams/main.py [hunk 3]
```diff
@@ -11,14 +11,96 @@ def tokenize(text: str) -> list[str] | None:
     :param text: a text
     :return: a list of lower-cased tokens without punctuation
     """
+    if isinstance(text, str):
```
Comment (artyomtugaryov) [inline lab_1_classify_by_unigrams/main.py:14]: try early return https://habr.com/ru/articles/348074/

File: lab_1_classify_by_unigrams/main.py [hunk 4]
```diff
@@ -11,14 +11,96 @@ def tokenize(text: str) -> list[str] | None:
...
+        text = [i for i in text.lower() if i.isalpha()]
```
Comment (artyomtugaryov) [inline lab_1_classify_by_unigrams/main.py:15]: I think, you do not need to create a variable - just return directly

File: lab_1_classify_by_unigrams/main.py [hunk 5]
```diff
@@ -11,14 +11,96 @@ def tokenize(text: str) -> list[str] | None:
...
 
+    tokenize(text)
 
 def calculate_frequencies(tokens: list[str] | None) -> dict[str, float] | None:
     """
     Calculates frequencies of given tokens
     :param tokens: a list of tokens
     :return: a dictionary with frequencies
     """
+    tokens = ['v', 'o', 'n', 'f', 'r', 'i', 's', 'c', 'h', 'n', 'o', 't', 'e', 'd', 's', 'o', 'm', 'e', 't', 'h', 'i',
```
Comment (artyomtugaryov) [inline lab_1_classify_by_unigrams/main.py:28]: hope it is for debug

File: lab_1_classify_by_unigrams/main.py [hunk 6]
```diff
@@ -11,14 +11,96 @@ def tokenize(text: str) -> list[str] | None:
...
+              'r', 'i', 's', 'c', 'h', 's', 'r', 'e', 'm', 'a', 'r', 'k', 'a', 'b', 'l', 'e', 'd', 'i', 's', 'c', 'o',
+              'v', 'e', 'r', 'i', 'e', 's', 's', 'o', 'o', 'n', 'h', 'e', 'w', 'o', 'u', 'l', 'd', 'a', 'l', 's', 'o',
+              'd', 'i', 's', 'c', 'o', 'v', 'e', 'r', 'a', 'n', 'u', 'm', 'b', 'e', 'r', 'o', 'f', 'o', 't', 'h', 'e',
+              'r', 'r', 'e', 'm', 'a', 'r', 'k', 'a', 'b', 'l', 'e', 'f', 'a', 'c', 't', 's', 'a', 'b', 'o', 'u', 't',
+              'h', 'o', 'w', 'b', 'e', 'e', 's', 'c', 'o', 'm', 'm', 'u', 'n', 'i', 'c', 'a', 't', 'e', 'a', 'n', 'd',
+              'i', 'n', 'd', 'o', 'i', 'n', 'g', 's', 'o', 'r', 'e', 'v', 'o', 'l', 'u', 't', 'i', 'o', 'n', 'i', 's',
+              'e', 't', 'h', 'e', 's', 't', 'u', 'd', 'y', 'o', 'f', 'a', 'n', 'i', 'm', 'a', 'l', 'b', 'e', 'h', 'a',
+              'v', 'i', 'o', 'u', 'r', 'g', 'e', 'n', 'e', 'r', 'a', 'l', 'l', 'y']
+
+    if isinstance(tokens, list):
```
Comment (artyomtugaryov) [inline lab_1_classify_by_unigrams/main.py:82]: again - early return

**Review by artyomtugaryov — CHANGES_REQUESTED** [review]

File: lab_1_classify_by_unigrams/main.py (hunk 3 above)
Comment (kobyakovapolina) [reply lab_1_classify_by_unigrams/main.py:14]: Fixed

File: lab_1_classify_by_unigrams/main.py (hunk 6 above)
Comment (kobyakovapolina) [reply lab_1_classify_by_unigrams/main.py]: Fixed

File: lab_1_classify_by_unigrams/main.py (hunk 5 above)
Comment (kobyakovapolina) [reply lab_1_classify_by_unigrams/main.py]: Removed it

**Comment by kobyakovapolina** [comment]
I have a cross in front of PR name check. What can I do to fix this?

File: lab_1_classify_by_unigrams/main.py [hunk 7]
```diff
@@ -28,7 +42,14 @@ def create_language_profile(language: str, text: str) -> dict[str, str | dict[st
     :param text: a text
     :return: a dictionary with two keys – name, freq
     """
-
+    if not isinstance(language, str) or not isinstance(text, str):
+        return None
+    else:
```
Comment (artyomtugaryov) [inline lab_1_classify_by_unigrams/main.py:47]: .

File: lab_1_classify_by_unigrams/start.py [hunk 8]
```diff
@@ -1,6 +1,8 @@
 """
 Language detection starter
 """
+#from lab_1_classify_by_unigrams.main import create_language_profile
```
Comment (artyomtugaryov) [inline lab_1_classify_by_unigrams/start.py:4]: remove

Comment (kobyakovapolina) [reply lab_1_classify_by_unigrams/start.py:4]: Removed

File: lab_1_classify_by_unigrams/main.py [hunk 9]
```diff
@@ -11,15 +11,29 @@ def tokenize(text: str) -> list[str] | None:
...
 def calculate_frequencies(tokens: list[str] | None) -> dict[str, float] | None:
     """
     Calculates frequencies of given tokens
     :param tokens: a list of tokens
     :return: a dictionary with frequencies
     """
-
+    if not isinstance(tokens, list) or not all([isinstance(i, str) for i in tokens]):
+        return None
+    else:
```
[the remark this answers is not in the document]
Comment (kobyakovapolina) [reply lab_1_classify_by_unigrams/main.py:27]: Fixed

File: lab_1_classify_by_unigrams/main.py (hunk 7 above)
Comment (kobyakovapolina) [reply lab_1_classify_by_unigrams/main.py:47]: Fixed

File: lab_1_classify_by_unigrams/main.py [hunk 10]
```diff
@@ -11,15 +11,29 @@ def tokenize(text: str) -> list[str] | None:
...
+    if not isinstance(tokens, list) or not all([isinstance(i, str) for i in tokens]):
+        return None
 
+    tokens_count = len(tokens)
+    list_tokens = {}
+    for i in tokens:
+        if i in list_tokens:
+            list_tokens[i] += 1
+        else:
+            list_tokens[i] = 1
```
Comment (artyomtugaryov) [inline lab_1_classify_by_unigrams/main.py:34]: Let set 0 if i is not in tokens and always do += 1

**Review by artyomtugaryov — CHANGES_REQUESTED** [review]

File: lab_1_classify_by_unigrams/main.py (hunk 10 above)
Comment (kobyakovapolina) [reply lab_1_classify_by_unigrams/main.py:34]: Done

**Pull request closed without merging**